acceptance/tests/connect/connect_proxy_lifecycle_test.go

```diff
@@ -212,3 +213,141 @@ func TestConnectInject_ProxyLifecycleShutdown(t *testing.T) {
 		})
 	}
 }
+
+func TestConnectInject_ProxyLifecycleShutdownJob(t *testing.T) {
+	cfg := suite.Config()
+	ver, err := version.NewVersion("1.2.0")
+	require.NoError(t, err)
+	if cfg.ConsulDataplaneVersion != nil && cfg.ConsulDataplaneVersion.LessThan(ver) {
+		t.Skipf("skipping this test because proxy lifecycle management is not supported in consul-dataplane version %v", cfg.ConsulDataplaneVersion.String())
+	}
+	cases := map[string]struct {
+		secure      bool
+		gracePeriod int64
+	}{
+		"Insecure with grace period":    {false, 30},
+		"Insecure without grace period": {false, 0},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			ctx := suite.Environment().DefaultContext(t)
+			releaseName := helpers.RandomName()
+
+			connHelper := connhelper.ConnectHelper{
+				ClusterKind: consul.Helm,
+				Secure:      tc.secure,
+				ReleaseName: releaseName,
+				Ctx:         ctx,
+				Cfg:         cfg,
+				HelmValues: map[string]string{
+					"connectInject.sidecarProxy.lifecycle.defaultShutdownGracePeriodSeconds": strconv.FormatInt(tc.gracePeriod, 10),
+					"connectInject.sidecarProxy.lifecycle.defaultEnabled":                    strconv.FormatBool(tc.gracePeriod > 0),
+				},
+			}
+
+			connHelper.Setup(t)
+			connHelper.Install(t)
+			connHelper.DeployJobAndServer(t)
```

Comment:
I think this test needs to be slightly different as you are shouldn't need to install so many times. I am fine if this test only runs for the insecure scenario.

1) Install server only. Split DeployJobAndServer() into DeployServer() and DeployJob(name string).
2) Deploy job 1 (with override graceful annotation X), check connection, check shutdown
3) Deploy job2 (with override graceful annotation Y), check connection, check shutdown
4) Deploy job3 (with override graceful annotation Z), check connection, check shutdown